### PR TITLE
MOM6 backgrounds in cycled DA mode were hardcoded for IAU 

### LINKED ISSUE #1352: Missing gdas.t??z.ocnf009.nc when cycling S2S at C384O025 with no iau
**Expected behavior**
At the end of a cycle, we should have 3 ocean backgrounds, for ex: `gdas.t00z.ocnf003.nc`, `gdas.t00z.ocnf006.nc` and `gdas.t00z.ocnf009.nc`. 

**Current behavior**
`gdas.t00z.ocnf009.nc` is missing

### LINKED ISSUE #1359: ocnf00* have the wrong dates
**Expected behavior**
`ocnf003` : beginning of the DA window
`ocnf006` : middle       "     "
`ocnf009` : end       "     "      

**Current behavior**
These files above are off by 3 hours.

**Machines affected**
all

**To Reproduce**
run the cycled forecast

### PR DESCRIPTION
**Pull request by guillaumevernieres**
### Description
The ocean backgrounds linked to COMROT were wrong when not doing IAU. This is mainly a copy/paste work of what's done for the atmosphere.

### Testing
Tested in cycled/da mode at c48o500 and c384o025 

- fixes #1359

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
humm ... I was a bit trigger happy on that one, sorry. I clicked submit before the high-res test was done ... and of course it's wrong. I don't understand why the low-res would work but not the high-res. 
Closing for now since this work isn't fixing the issue. Oops.

**Comment by guillaumevernieres** [comment]
No clue why MOM6 is not dumping the history at the end. This PR still does fix the iau/no iau bug.

File: ush/forecast_postdet.sh [hunk 1]
```diff
@@ -876,15 +876,13 @@ MOM6_postdet() {
   elif [[ "${CDUMP}" =~ "gdas" ]]; then
     # Link output files for CDUMP = gdas
 
-    # MOM6 does not write out the first forecast hour, so start at FHOUT
-    local fhr="${FHOUT}"
-    while [[ "${fhr}" -le "${FHMAX}" ]]; do
+    # Save MOM6 backgrounds
+    local fhr="${FHMIN}"
```
Comment (aerorahul) [inline ush/forecast_postdet.sh]: this line is unnecessary.
```suggestion
```

File: ush/forecast_postdet.sh [hunk 2]
```diff
@@ -876,15 +876,13 @@ MOM6_postdet() {
...
+    for fhr in ${OUTPUT_FH}; do
```
Comment (aerorahul) [inline ush/forecast_postdet.sh:880]: Technically, `OUTPUT_FH` is only valid for atmosphere output.  That was the reason for the while loop.  If this works for you right now, then its fine for now.
MOM6 output frequency is controlled differently.

**Review by aerorahul — COMMENTED** [review]
Looks ok.  Please commit the suggestion.

**Review by WalterKolczynski-NOAA — COMMENTED** [review]
Agree with Rahul's suggestion and confirm his comment. Will approve after the change has been made.

**Review by aerorahul — APPROVED** [review]
Looks good.

**Comment by guillaumevernieres** [comment]
Thanks for the reviews/code cleanup @aerorahul and @WalterKolczynski-NOAA .

**Pull request merged**